src/mempool/mempool.rs

```diff
@@ -172,19 +281,22 @@ impl Mempool {
                 }
                 message = self.mempool_rx.recv() => {
                     if let Some(message) = message {
-                        // TODO(aditi): Maybe we don't need to run validations here?
-                        if self.message_is_valid(&message) {
-                            let fid = message.fid();
-                            let shard_id = self.message_router.route_message(fid, self.num_shards);
-                            // TODO(aditi): We need a size limit on the mempool and we need to figure out what to do if it's exceeded
-                            match self.messages.get_mut(&shard_id) {
-                                None => {
-                                    let mut messages = BTreeMap::new();
-                                    messages.insert(MempoolKey { inserted_at: Instant::now()}, message.clone());
-                                    self.messages.insert(shard_id, messages);
+                        self.insert(message).await;
+                    }
+                }
+                chunk = self.shard_decision_rx.recv() => {
```

Comment:
Is there a reason to use the shard decision rx over the block decision rx?

Reply:
I did this at the block level at first, but @sanjayprabhu recommended doing it at the shard level instead here: https://github.com/farcasterxyz/snapchain/pull/200#discussion_r1908215651

Reply:
Got it, that reasoning makes sense. There are minimal chances that a message is committed at the shard level but not the block level.